### PR TITLE
Mz ce view purchase details

### PR DESCRIPTION
**Pull request by mindyzwan**
- Added a new view, `ItemDetails`, that lists out the details of an individual item. 
- Used CSS to show/hide a view more button on hover. 
- Used react router dom's `useParams()` function and the colon syntax in the router to go to pages based off of specific item IDs. 

**AC:**
From the shopping list, user can tap to enter a detailed view of the item 

Detail view includes the following information: 
- Name of the item 
- Last purchased date 
- Estimated next purchase date 
- Number of times the item has been purchased

Accessibility for item details view is good! Though we worked in the fetch items page it made more sense to fix those accessibility issues in a separate PR after this week's AC's get merged. 

Let us know what you think!

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -93,26 +97,43 @@ a:visited {
...
 
-.nonPurchasedItem {
+.listItem {
   list-style-type: none;
+  width: fit-content;
+  margin: 5vh auto;
+  display: flex;
+}
+
+.listItem:hover .viewMore {
```
Comment (michellejae) [inline src/App.css]: NOIIIICE

File: cypress/integration/item_details.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+describe('Item details view', function() {
+  beforeEach(function() {
+		window.localStorage.setItem('uniqueToken', 'token1234')
+		const testDate = new Date('2019-01-01');
+		cy.addItem('token1234', 'banana', testDate);
+  });
+	
+	it('Shows current date as last purchase date when purchased today', function() {
+		let todayString = new Date().toDateString();
```
[the remark this answers is not in the document]
Comment (mindyzwan) [reply cypress/integration/item_details.js]: Done! :)

File: cypress/support/commands.js [hunk 3]
```diff
@@ -23,3 +23,21 @@
...
+			name: itemID,
+			numberOfDays: 14,
+			dateOfPurchase: dateOfPurchase,
+			numberOfPurchases: 1,
+	})
+});
+
+Cypress.Commands.add("deleteItem", (token, itemID) => { 
+	db.collection("lists").doc(token).collection('items').doc(itemID).delete();
+ })
```
[the remark this answers is not in the document]
Comment (mindyzwan) [reply cypress/support/commands.js]: Basically, yeah! I was trying to figure out where to put the functions I needed to delete/add items and ran across the custom commands section. I played around with it and was excited to see it work!

File: src/component/ItemDetails.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+const ItemDetails = ({ token, setToken }) => {
+  let { itemId } = useParams();
+  const concatPath = `/lists/${token}/items`;
+
+  return (
+    <FirestoreCollection
+      path={concatPath}
+      render={({ isLoading, data }) => {
+        // Finds specific item where the url param :itemId is equal to the item ID in the list.
+        const item = data.find(x => x.id === itemId);
```
[the remark this answers is not in the document]
Comment (mindyzwan) [reply src/component/ItemDetails.js]: Noted and updated!

File: src/component/ItemDetails.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+        if (isLoading) {
+          return <div>Still Loading...</div>;
+        } else {
+          // Below lines calculate next purchase date using dayjs 'add' function.
+          const lastPurchaseDate = item.dateOfPurchase.toDate();
```
[the remark this answers is not in the document]
Comment (mindyzwan) [reply src/component/ItemDetails.js]: I went ahead and changed it! 👍

**Pull request merged**